src/parser/content-evaluator-module.ts

```diff
@@ -48,34 +49,73 @@ export class ContentEvaluatorModule implements Module {
 
   async _processComment(comments: Readonly<GithubCommentScore>[], specificationBody: string) {
     const commentsWithScore: GithubCommentScore[] = [...comments];
-    const commentsBody = commentsWithScore.map((comment) => comment.content);
-    const relevance = await this._evaluateComments(specificationBody, commentsBody);
 
-    if (relevance.length !== commentsWithScore.length) {
-      console.error("Relevance / Comment length mismatch! Skipping.");
-      return [];
+    const fixedRelevances: { [k: string]: number } = this._relevancesForCommentTypes();
+
+    // exclude comments that have fixed relevance multiplier. e.g. review comments = 1
+    const commentsToEvaluate: { id: number; comment: string }[] = [];
+    for (let i = 0; i < commentsWithScore.length; i++) {
+      const currentComment = commentsWithScore[i];
+      if (!fixedRelevances[currentComment.type]) {
+        commentsToEvaluate.push({
+          id: currentComment.id,
+          comment: currentComment.content,
+        });
+      }
     }
 
-    for (let i = 0; i < relevance.length; i++) {
+    const relevancesByAI = await this._evaluateComments(specificationBody, commentsToEvaluate);
+
+    if (Object.keys(relevancesByAI).length !== commentsToEvaluate.length) {
+      console.error("Relevance / Comment length mismatch! \nWill use 1 as relevance for missing comments.");
+    }
+
+    for (let i = 0; i < commentsWithScore.length; i++) {
       const currentComment = commentsWithScore[i];
-      const currentRelevance = relevance[i];
+      let currentRelevance = 1; // For comments not in fixed relevance types or missed by OpenAI evaluation
+
+      if (fixedRelevances[currentComment.type]) {
+        currentRelevance = fixedRelevances[currentComment.type];
+      } else if (!isNaN(relevancesByAI[currentComment.id])) {
+        currentRelevance = relevancesByAI[currentComment.id];
+      }
+
       const currentReward = new Decimal(currentComment.score?.reward || 0);
       currentComment.score = {
         ...(currentComment.score || {}),
-        relevance: currentRelevance.toNumber(),
+        relevance: new Decimal(currentRelevance).toNumber(),
         reward: currentReward.mul(currentRelevance).toNumber(),
       };
     }
 
     return commentsWithScore;
   }
 
-  async _evaluateComments(specification: string, comments: string[]): Promise<Decimal[]> {
+  _relevancesForCommentTypes(): { [k: string]: number } {
+    const relevances: { [k: string]: number } = {};
+    this._configuration.multipliers.forEach((multiplier) => {
+      let commentType = 0;
+      multiplier.targets.forEach((target) => {
+        commentType |= CommentType[target as keyof typeof CommentType];
+      });
+
+      if (multiplier.relevance !== undefined) {
+        relevances[commentType as unknown as string] = multiplier.relevance;
```

Comment:
Why is casting to string needed?

Reply:
I would imagine that relevance scoring should be a number.

Reply:
> Why is casting to string needed?

Yeah that was dumb of me. Some leftovers of strickly typed languages in my head. Resolved by removing the casting. 

> I would imagine that relevance scoring should be a number.

correct. a floating point number. whilefoo is referring to the key here.